src/app/share/views/SharedGuestSignUp/ShareGuestSingUpView.tsx

```diff
@@ -0,0 +1,306 @@
+    }
+  }, [invitationId]);
+
+  useEffect(() => {
+    if (user && mnemonic) {
+      dispatch(userActions.setUser(user));
+      if (user?.registerCompleted && mnemonic) {
+        return navigationService.push(AppView.Shared);
+      }
+    }
+  }, []);
+
+  useEffect(() => {
+    if (password.length > 0) onChangeHandler(password);
+  }, [password]);
+
+  function onChangeHandler(input: string) {
+    setIsValidPassword(false);
+    if (input.length > MAX_PASSWORD_LENGTH) {
+      setPasswordState({ tag: 'error', label: translate('modals.changePasswordModal.errors.longPassword') });
+      return;
+    }
+
+    const result = testPasswordStrength(input, (qs.email as string) === undefined ? '' : (qs.email as string));
+
+    setIsValidPassword(result.valid);
+    if (!result.valid) {
+      setPasswordState({
+        tag: 'error',
+        label:
+          result.reason === 'NOT_COMPLEX_ENOUGH'
+            ? 'Password is not complex enough'
+            : 'Password has to be at least 8 characters long',
+      });
+    } else if (result.strength === 'medium') {
+      setPasswordState({ tag: 'warning', label: 'Password is weak' });
+    } else {
+      setPasswordState({ tag: 'success', label: 'Password is strong' });
+    }
+  }
+
+  const onSubmit: SubmitHandler<IFormValues> = async (formData, event) => {
+    event?.preventDefault();
+    setIsLoading(true);
+
+    try {
+      const { email, password, token } = formData;
+      const { xUser, xToken, mnemonic } = await doRegisterPreCreatedUser(email, password, invitationId ?? '', token);
+
+      localStorageService.removeItem(STORAGE_KEYS.SIGN_UP_TUTORIAL_COMPLETED);
+
+      localStorageService.clear();
+
+      localStorageService.set('xToken', xToken);
+      localStorageService.set('xMnemonic', mnemonic);
+
+      const xNewToken = await getNewToken();
+      localStorageService.set('xNewToken', xNewToken);
+
+      const decryptedPrivateKey = decryptPrivateKey(xUser.privateKey, password);
+
+      const privateKey = xUser.privateKey ? Buffer.from(decryptedPrivateKey).toString('base64') : undefined;
+
+      const user = {
+        ...xUser,
+        privateKey,
+      } as UserSettings;
+
+      dispatch(userActions.setUser(user));
+      await dispatch(userThunks.initializeUserThunk());
+      dispatch(productsThunks.initializeThunk());
+      dispatch(planThunks.initializeThunk());
+
+      dispatch(referralsThunks.initializeThunk());
+
+      window.rudderanalytics.identify(xUser.uuid, { email, uuid: xUser.uuid });
+      window.rudderanalytics.track('User Signup', { email });
+
+      return navigationService.push(AppView.Shared);
+    } catch (err: unknown) {
+      setIsLoading(false);
+      errorService.reportError(err);
+      const castedError = errorService.castError(err);
+      setSignupError(castedError.message);
+    } finally {
+      setShowError(true);
+    }
+  };
+
+  if (invitationValidation.isLoading) {
+    return <></>;
```

Comment:
Any reason not to add a charging element, such as a spinner or any kind of element that indicates that it is loading?

Reply:
I created a thread in slack mentioning we needed this like 2 weeks ago, and one of the possible options is to show an empty page while this verification is being made as the backend returns a quick response. 

If you want another component to be shown, please go ahead and let me know.